File: scala/cookieclicker/src/main/scala/com/example/Cookie.scala
package cookieclicker

import akka.actor.{Actor, ActorRef}
import Score._

class Cookie(score: ActorRef) extends Actor {
    import Cookie._
    override def receive: Receive = {
        case Click => score ! Increase
    }
}

object Cookie {
    def apply(score: ActorRef): Cookie = {
        new Cookie(score)
    }
    case class Click()
}

File: csharp/CookieClickerGame/Cookie.cs
using Akka.Actor;
using System;
using System.Collections.Generic;
using System.Text;

namespace CookieClickerGame
{
    public class Cookie : ReceiveActor
    {
        #region Messages
        public class Click { }
        #endregion
        public IActorRef Score { get; }

        public Cookie(IActorRef score)
        {
            Receive<Click>(HandleClick);
            Score = score;
        }

        private void HandleClick(Click obj)
        {
            Score.Tell(new Score.Increase());
        }
    }
}
